### PR TITLE
feat: increase log level when authentication failed

### PR DESCRIPTION
**Pull request by zoumo**
reduce some noisy log

### CODE REVIEW

File: pkg/server/config.go [hunk 1]
```diff
@@ -269,6 +270,12 @@ type SecureServingInfo struct {
 
 	// DisableHTTP2 indicates that http2 should not be enabled.
 	DisableHTTP2 bool
+
+	// ErrorLog specifies an optional logger for errors accepting
+	// connections, unexpected behavior from handlers, and
+	// underlying FileSystem errors.
+	// If nil, logging is done via the log package's standard logger.
+	ErrorLog *log.Logger
```
Comment (xuchen-xiaoying) [inline pkg/server/config.go:278]: I didn't find usage of this field

File: pkg/endpoints/filters/authentication.go [hunk 2]
```diff
@@ -50,15 +50,15 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 		defer recordAuthMetrics(resp, ok, err, apiAuds, authenticationStart)
 		if err != nil || !ok {
 			if err != nil {
-				klog.Errorf("Unable to authenticate the request due to an error: %v", err)
+				klog.V(5).Infof("Unable to authenticate the request due to an error: %v", err)
```
Comment (xuchen-xiaoying) [inline pkg/endpoints/filters/authentication.go]: I think that auth err log is necessary, do we need to adjust log level?

File: pkg/server/config.go (hunk 1 above)
Comment (zoumo) [reply pkg/server/config.go:278]: It will be used in kube-gateway code, not here

File: pkg/endpoints/filters/authentication.go (hunk 2 above)
Comment (zoumo) [reply pkg/endpoints/filters/authentication.go]: There are too many logs here when gateway receives invalid token.
But after thinking deeply, I will revert this file now. It seems that we should reduce the number of requests with invaid token firstly

**Review by xuchen-xiaoying — APPROVED** [review]
LGTM

**Pull request merged**